src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -538,21 +539,26 @@ public void acceptRecord(final SAMRecord record) {
                         boolean isOnTarget = false;
                         for (final Interval target : targets) {
                             if (refPos >= target.getStart() && refPos <= target.getEnd()) {
-                                isOnTarget = true;
                                 ++this.metrics.ON_TARGET_BASES;
                                 if (mappedInPair) ++this.metrics.ON_TARGET_FROM_PAIR_BASES;
 
                                 final int targetOffset = refPos - target.getStart();
                                 final Coverage coverage = this.coverageByTarget.get(target);
                                 coverage.addBase(targetOffset);
                                 baseQHistogramArray[baseQualities[offset]]++;
+                                coveredTargets.add(target);
```

Comment:
if you add the line
```
if (!coveredTargets.contains(target)) coverage.incrementReadCount();
``` 
above `coveredTargets.add(target);` then I think you can get rid of the for loop below.